### ISSUE TITLE
appStyles tree merge happens too late

### ISSUE THREAD

**Issue by ef4** [embroider-build/embroider#23]
We need to merge the appStyles trees from all classic addons _before_ we try to transpile the app's stylesheets. Because they might be trying to import from each other.

**Comment by knownasilya** [comment]
I get an error on `ember serve` in an app. The app uses an addon that has org in the name, e.g. `@org/someaddon` and that addon exports sass styles.

```
File not found: /addon.scss
in any of the following include paths:
  /Users/ilya/workspace/build/tmp/sass_compiler-input_base_path-14F44Vcb.tmp
        at OneShot
-~- created here: -~-
    at new Plugin (/Users/ilya/workspace/build/node_modules/broccoli-plugin/index.js:7:31)
    at new OneShot (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/one-shot.js:14:9)
    at Object.cachedBuildCompatAddon [as default] (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/build-compat-addon.js:15:16)
    at CompatAddons.movedAddons.map.oldPkg (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/compat-addons.js:33:105)
    at Array.map (<anonymous>:null:null)
    at new CompatAddons (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/compat-addons.js:33:63)
    at Object.defaultPipeline [as compatBuild] (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/default-pipeline.js:11:18)
    at module.exports (/Users/ilya/workspace/build/ember-cli-build.js:50:39)
    at Builder.readBuildFile (/Users/ilya/workspace/build/node_modules/ember-cli/lib/models/builder.js:52:14)
    at Builder.setupBroccoliBuilder (/Users/ilya/workspace/build/node_modules/ember-cli/lib/models/builder.js:66:22)
    at new Builder (/Users/ilya/workspace/build/node_modules/ember-cli/lib/models/builder.js:32:10)
    at ServeTask.run (/Users/ilya/workspace/build/node_modules/ember-cli/lib/tasks/serve.js:45:55)
    at Promise.resolve.then (/Users/ilya/workspace/build/node_modules/ember-cli/lib/models/command.js:243:46)
    at tryCatcher (/Users/ilya/workspace/build/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/Users/ilya/workspace/build/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
    at /Users/ilya/workspace/build/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:559:14
    at flush (/Users/ilya/workspace/build/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2402:5)
    at processTicksAndRejections (internal/process/next_tick.js:74:9)
```

**Comment by knownasilya** [comment]
Hum, I think that might have been caused because someaddon was yarn linked.

Now I get a different error.

```
 - broccoliBuilderErrorStack: Error: Cannot find module 'ember-data/package.json' from '/Users/ilya/workspace/build/node_modules/ember-decorators'
    at Function.module.exports [as sync] (/Users/ilya/workspace/build/node_modules/resolve/lib/sync.js:58:15)
    at get_or_create_1.getOrCreate (/Users/ilya/workspace/build/node_modules/@embroider/core/src/package-cache.js:20:57)
    at Object.getOrCreate (/Users/ilya/workspace/build/node_modules/@embroider/core/src/get-or-create.js:6:18)
    at MovablePackageCache.resolve (/Users/ilya/workspace/build/node_modules/@embroider/core/src/package-cache.js:19:32)
    at names.map.name (/Users/ilya/workspace/build/node_modules/@embroider/core/src/package.js:76:52)
    at Array.map (<anonymous>:null:null)
    at TweakedPackage.get dependencies (/Users/ilya/workspace/build/node_modules/@embroider/core/src/package.js:76:22)
    at TweakedPackage.<anonymous> (/Users/ilya/workspace/build/node_modules/typescript-memoize/dist/memoize-decorator.js:67:52)
    at DependencyAnalyzer.get externals [as externals] (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/dependency-analyzer.js:26:34)
    at RewritePackageJSON.build (/Users/ilya/workspace/build/node_modules/@embroider/compat/src/rewrite-package-json.js:30:40)
    at /Users/ilya/workspace/build/node_modules/broccoli-plugin/read_compat.js:93:34
    at tryCatch (/Users/ilya/workspace/build/node_modules/rsvp/dist/rsvp.js:525:12)
    at invokeCallback (/Users/ilya/workspace/build/node_modules/rsvp/dist/rsvp.js:538:13)
    at publish (/Users/ilya/workspace/build/node_modules/rsvp/dist/rsvp.js:508:7)
    at flush (/Users/ilya/workspace/build/node_modules/rsvp/dist/rsvp.js:2415:5)
    at processTicksAndRejections (internal/process/next_tick.js:74:9)
```

**Comment by ef4** [comment]
Please file a separate issue for that, it's not related to this one.

**Comment by knownasilya** [comment]
Removed the ember-data issue since #48 covers that. Making a new issue for `yarn link`ed addons.

**Comment by ef4** [comment]
This was fixed in 0.3.1.
